src/backend/libs-and-metadata.md

```diff
@@ -114,21 +114,22 @@ computed.
 [incremental compilation]: ../queries/incremental-compilation.md
 [`finalize_and_compute_crate_hash`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/hir/map/collector/struct.NodeCollector.html#method.finalize_and_compute_crate_hash
 
-### Crate Disambiguator
+### Stable Crate Id
 
-The [`CrateDisambiguator`] is a 128-bit hash used to distinguish between
-different crates of the same name. It is a hash of all the [`-C metadata`] CLI
-options computed in [`compute_crate_disambiguator`]. It is used in a variety
-of places, such as symbol name mangling, crate loading, and much more.
+The [`StableCrateId`] is a 64-bit hash used to identify different crates with
+potentially the same name. It is a hash of the crate name and all the
+[`-C metadata`] CLI options computed in [`StableCrateId::new`]. It is
+used in a variety of places, such as symbol name mangling, crate loading, and
+much more.
 
-By default, all Rust symbols are mangled and incorporate the disambiguator
-hash. This allows multiple versions of the same crate to be included together.
-Cargo automatically generates `-C metadata` hashes based on a variety of
-factors, like the package version, source, and the target kind (a lib and bin
-can have the same crate name, so they need to be disambiguated).
+By default, all Rust symbols are mangled and incorporate the stable crate id.
+This allows multiple versions of the same crate to be included together. Cargo
+automatically generates `-C metadata` hashes based on a variety of factors,
+like the package version, source, and the target kind (a lib and test can have
+the same crate name, so they need to be disambiguated).
```

Comment:
If the bin crate type is used, this is included in the hash and as such a lib and bin never have the same hash. The comment about this describes this exact use case. This was true even before `CrateDisambiguator` was merged into `StableCrateId`.